### PR TITLE
JSUI-3265 Remove Sharepoint (actually Angular) work-around

### PR DESCRIPTION
**Pull request by samisayegh**
I learned on Friday that the leading `/` was not introduced by Sharepoint, but rather the Angular framework being used to build the app. The Intuit team has fixed the behavior by setting an option on the framework.

In theory, this means we the work-around in the JSUI is no longer needed. This PR removes the lines.

Or do you prefer to keep the logic? If so, then I will rename Sharepoint to Angular for clarity,

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

**Comment by olamothe** [comment]
I mean.. is there a lot of harm of keeping it there ? Assuming it works well... ?

**Comment by samisayegh** [comment]
There is no harm in keeping the code. The main argument might be the slight difference in behaviour with the url in the AuthenticationProvider compared to the other parts of the JSUI interacting with the url.

If the preference is to keep the code, I will just rename `Sharepoint` -> `Angular` for clarity for future maintainers,

**Review by ThibodeauJF — APPROVED** [review]

**Review by olamothe — APPROVED** [review]
I guess I'll leave it up to your judgement: Do you think the code is more (or less) solid with this removed or not ?

**Pull request merged**
